src/main/java/com/devonfw/quarkus/productmanagement/domain/repo/ProductFragmentImpl.java

```diff
@@ -28,87 +21,33 @@
   EntityManager em;
 
   @Override
-  public Page<ProductEntity> findAllCriteriaApi(ProductSearchCriteriaDto dto) {
-
-    CriteriaQuery<ProductEntity> cq = this.em.getCriteriaBuilder().createQuery(ProductEntity.class);
-    Root<ProductEntity> root = cq.from(ProductEntity.class);
-    List<Predicate> predicates = new ArrayList<>();
-    CriteriaBuilder cb = this.em.getCriteriaBuilder();
-    if (dto.getTitle() != null && !dto.getTitle().isEmpty()) {
-      predicates.add(cb.like(root.get(ProductEntity_.TITLE), dto.getTitle()));
-    }
-
-    if (dto.getPrice() != null) {
-      predicates.add(cb.gt(root.get(ProductEntity_.PRICE), dto.getPrice()));
-    } else if (dto.getPriceMin() != null || dto.getPriceMax() != null) {
-      if (dto.getPriceMin() != null) {
-        predicates.add(cb.gt(root.get(ProductEntity_.PRICE), dto.getPriceMin()));
-      }
-      if (dto.getPriceMax() != null) {
-        predicates.add(cb.lt(root.get(ProductEntity_.PRICE), dto.getPriceMax()));
-      }
-    }
-
-    if (!predicates.isEmpty())
-
-    {
-      cq.where(predicates.toArray(new Predicate[0]));
-    }
-
-    // Order by title
-    cq.orderBy(cb.desc(root.get(ProductEntity_.TITLE)));
-
-    TypedQuery<ProductEntity> products = this.em.createQuery(cq).setFirstResult(dto.getPageNumber() * dto.getPageSize())
-        .setMaxResults(dto.getPageSize());
-    return new PageImpl<ProductEntity>(products.getResultList(), PageRequest.of(dto.getPageNumber(), dto.getPageSize()),
-        products.getResultList().size());
-  }
-
-  @Override
-  public Page<ProductEntity> findAllQueryDsl(ProductSearchCriteriaDto dto) {
+  public Page<ProductEntity> findProducts(ProductSearchCriteriaDto searchCriteria) {
 
     QProductEntity product = QProductEntity.productEntity;
-    JPAQuery<ProductEntity> query = new JPAQuery<ProductEntity>(this.em);
-    query.from(product);
-    if (dto.getTitle() != null && !dto.getTitle().isEmpty()) {
-      query.where(product.title.eq(dto.getTitle()));
+    JPAQuery<ProductEntity> query = new JPAQuery<ProductEntity>(this.em).from(product);
+
+    if (!StringUtils.isEmpty(searchCriteria.getTitle())) {
```

Comment:
this implementation only allows restriction based on the last criteria that is present as `query.where` replaces the previously added restriction(s). This is not the expected behaviour from users point of view - e.g. if i set both min and maxprice, I expect to get results where  `minPrice < product.price > maxPrice`
Standard approach is to collect the where clause predicates in some collection and then pass them as arg to `query.where` at the end.

Reply:
Done.

Reply:
This is IMHO wrong. All calls to `where` method will be combined with `AND`.